components/style/properties/longhand/inherited_box.mako.rs

```diff
@@ -50,6 +50,132 @@ ${helpers.single_keyword("image-rendering",
+            FromImage,
+            AngleWithFlipped(Angle, bool),
+        }
+    }
+
+    const INITIAL_ANGLE: Angle = Angle(0f32);
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::AngleWithFlipped(INITIAL_ANGLE, false)
+    }
+
+    use values::CSSFloat;
+    use std::f32::consts::PI;
+    const TWO_PI: CSSFloat = 2f32*PI;
+
+    #[inline]
+    fn normalize_angle(angle: &Angle) -> Angle {
+        let radians = angle.radians();
+        let normalized_radians = ((radians + PI/2f32 % TWO_PI) + TWO_PI) % TWO_PI;
+        Angle::from_radians(normalized_radians)
+    }
+
+    impl ToComputedValue for SpecifiedValue {
+        type ComputedValue = computed_value::T;
+
+        #[inline]
+        fn to_computed_value(&self, _: &Context) -> computed_value::T {
+            if let Some(ref angle) = self.angle {
+                let normalized_angle = normalize_angle(angle);
+                computed_value::T::AngleWithFlipped(normalized_angle, self.flipped)
+            } else {
+                if self.flipped {
+                    computed_value::T::AngleWithFlipped(INITIAL_ANGLE, true)
+                } else {
+                    computed_value::T::FromImage
+                }
+            }
+        }
+
+        #[inline]
+        fn from_computed_value(computed: &computed_value::T) -> Self {
+            match *computed {
+                computed_value::T::FromImage => SpecifiedValue { angle: None, flipped: false },
+                computed_value::T::AngleWithFlipped(angle, flipped) =>
+                    SpecifiedValue { angle: Some(angle), flipped: flipped },
+            }
+        }
+    }
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            match *self {
+                computed_value::T::FromImage => dest.write_str("from-image"),
+                computed_value::T::AngleWithFlipped(angle, flipped) => {
+                    try!(angle.to_css(dest));
+                    if flipped {
+                        try!(dest.write_str(" flipped"));
+                    }
+                    Ok(())
+                },
+            }
+        }
+    }
+
+    pub fn parse(context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        if input.try(|input| input.expect_ident_matching("from-image")).is_ok() {
+            // Handle from-image
+            Ok(SpecifiedValue { angle: None, flipped: false })
+        } else if let Ok(angle) = input.try(|i| Angle::parse(context, i)) {
+            // Handle <angle> and <angle> flip
+            let flipped = input.try(|input| input.expect_ident_matching("flip")).is_ok();
+            Ok(SpecifiedValue { angle: Some(angle), flipped: flipped })
+        } else {
+            // Handle flip
+            input.expect_ident_matching("flip")
+            .map(|()| SpecifiedValue { angle: Some(INITIAL_ANGLE), flipped: true })
+        }
```

Comment:
We don't need to return a parse error for this. It will parse it as `0deg`. After this parsing, `fl` will be unconsumed from parse and will try to consume `;` but it'll find `fl` instead and fail.